### PR TITLE
Improve test coverage for animation.dart

### PR DESCRIPTION
**Pull request by abarth**
We now have 100% coverage of animation.dart and animation_controller.dart.
Also, add some basic tools for working with lcov files. These tools need much
more polish.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

**Comment by eseidelGoogle** [comment]
FYI @cbracken

File: dev/tools/coverage.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
```
Comment (Hixie) [inline dev/tools/coverage.dart]: put a comment here saying what this is at a high level

**Comment by Hixie** [comment]
LGTM

For extra credit I would go further with your matcher. Rename it to "isGoodObject" and have it check:
- [ ] toString fulfills the criteria you test today
- [ ] toString doesn't crash
- [ ] toString isn't empty
- [ ] hashCode doesn't crash, and returns a value distinct from null.hashCode
- [ ] `operator ==` doesn't crash, and returns false (and doesn't crash) when compared to a `new Object()`, and returns true when compared to itself.

**Pull request merged**